### PR TITLE
[ELY-769] Part 1: Groundwork for algorithm spec matching in credential sources/sets

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

**Comment by dmlloyd** [comment]
Travis-CI check failure seems bogus

File: src/main/java/org/wildfly/security/credential/source/CallbackHandlerCredentialSource.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        Assert.checkNotNullParam("callbackHandler", callbackHandler);
+        this.callbackHandler = callbackHandler;
+    }
+
+    public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential> credentialType, final String algorithmName, final AlgorithmParameterSpec parameterSpec) throws IOException {
+        return SupportLevel.POSSIBLY_SUPPORTED;
+    }
+
+    public <C extends Credential> C getCredential(final Class<C> credentialType, final String algorithmName, final AlgorithmParameterSpec parameterSpec) throws IOException {
+        if (PasswordCredential.class.isAssignableFrom(credentialType) && (algorithmName == null || algorithmName.equals(ClearPassword.ALGORITHM_CLEAR)) && parameterSpec == null) {
```
Comment (darranl) [inline src/main/java/org/wildfly/security/credential/source/CallbackHandlerCredentialSource.java]: Looking at both Peter and Pedro's PRs I was worried that handling these callbacks was starting to be duplicated as they both needed to solve the same problem for different reasons and in some cases handled subtly differently - so this should become the place we interact with the callbacks.

**Comment by dmlloyd** [comment]
This one is ready to go in now.  I can cope with the OTP stuff next week; these changes as they are now do not impact OTP.

**Comment by darranl** [comment]
Added a +1 - I think we may want to start the week with an Elytron tag and a Remoting tag.

**Pull request merged**